packages/amplify_core/lib/src/amplify_class.dart

```diff
@@ -0,0 +1,134 @@
+  /// be called before Amplify has been configured. Customers are expected
+  /// to check the configuration state by calling `Amplify.isConfigured`
+  ///
+  /// Throws AmplifyAlreadyConfiguredException if
+  /// this method is called after configure (e.g. during hot reload).
+  Future<void> addPlugin(AmplifyPluginInterface plugin);
+
+  /// Adds multiple plugins at the same time. Note: this method can only
+  /// be called before Amplify has been configured. Customers are expected
+  /// to check the configuration state by calling `Amplify.isConfigured`
+  Future<void> addPlugins(List<AmplifyPluginInterface> plugins) {
+    return Future.wait(plugins.map(addPlugin), eagerError: true);
+  }
+
+  /// Returns whether Amplify has been configured or not.
+  bool get isConfigured => _configCompleter.isCompleted;
+
+  /// A future when completes when Amplify has been successfully configured.
+  Future<AmplifyConfig> get asyncConfig => _configCompleter.future;
+
+  /// Configures Amplify with the provided configuration string.
+  /// **This method can only be called once**, after all the plugins
+  /// have been added and no plugin shall be added after amplify
+  /// is configured. Customers are expected to call `Amplify.isConfigured`
+  /// to check if their app is configured before calling this method.
+  ///
+  /// Throws AmplifyAlreadyConfiguredException if
+  /// this method is called again (e.g. during hot reload).
+  Future<void> configure(String configuration) async {
+    if (isConfigured) {
+      throw const AmplifyAlreadyConfiguredException(
+        'Amplify has already been configured and re-configuration is not supported.',
+        recoverySuggestion:
+            'Check if Amplify is already configured using Amplify.isConfigured.',
+      );
+    }
+
+    final AmplifyConfig amplifyConfig;
+    try {
+      final json = jsonDecode(configuration) as Map;
+      amplifyConfig = AmplifyConfig.fromJson(json.cast());
+    } on FormatException catch (e) {
+      throw AmplifyException(
+        'The provided configuration is not a valid json. Check underlyingException.',
+        recoverySuggestion:
+            'Inspect your amplifyconfiguration.dart and ensure that the string is proper json',
+        underlyingException: e.toString(),
+      );
+    }
+
+    await configurePlatform(configuration);
+    _configCompleter.complete(amplifyConfig);
+  }
+
+  /// Configures the platform-specific implementation of Amplify using the
+  /// registered [config].
+  @protected
+  Future<void> configurePlatform(String config);
+
+  /// Constructs a Core platform.
+  /// Internal Use Only
+  @protected
+  AmplifyClass.protected();
+
+  /// The instance of [AmplifyClass] to use.
+  ///
+  /// Defaults to the platform-specific implementation.
+  static AmplifyClass? instance;
+
+  /// The library version.
+  String get version => '0.5.0';
+
+  /// Resets the Amplify implementation, removing all traces of Amplify from
```

Comment:
Are we absolutely sure about ALL traces?

Reply:
Sure that we want to do that? I think so, I can't think of anything we'd want to hold onto.